### ISSUE TITLE
Syntax error for innocent comment

### ISSUE THREAD

**Issue by erezsh** [PyCQA/pylint#3305]
Reproduce by running pylint on this code:

```python
class A:
    name: str
    # type: B = None
```

### pylint --version output
```bash
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)]

```

**Comment by PCManticore** [comment]
Thanks for the report.

For context, this is caused by our use of `typed_ast`, which we use to parse type annotations. For this library, the annotation you used is invalid, leading to the error you are noticing in `pylint`. Here's an example of how `typed_ast` currently fails to parse this snippet of code:
```

In [1]: import typed_ast.ast3 as ast

In [2]: ast.parse(open('a.py').read())
  File "<unknown>", line 3
    # type: B = None
                   ^
SyntaxError: misplaced type annotation
```

Since we cannot control this behaviour of `typed_ast`, whenever we detect such invalid cases, we retry the parsing in `astroid` without type comment parsing, which should result in `pylint` not crashing any longer. This is fixed in https://github.com/PyCQA/astroid/commit/2cb6818ad5db25fd0f32a193ab90425e7dfcbee0 but will only be part of the next `pylint 2.5` release. If you can give it a go with the version from `master` for `pylint` and `astroid`, you'll notice it's properly working now.

**Comment by erezsh** [comment]
"Not crashing any longer" is a good start, but what exactly *is* going to happen? Will it throw an error message?

**Comment by PCManticore** [comment]
Won't crash and no error is going to be emitted as it's the fault of typed_ast in the first place.
